### PR TITLE
Add missing stripe.iinvoice fields

### PR DESCRIPTION
**Pull request by dozoisch**
- Added: amount_paid, amount_remaining, billing, billing_reason, due_date, hosted_invoice_url, invoice_pdf, number

---

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://stripe.com/docs/api#invoice_object
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/stripe/index.d.ts [hunk 1]
```diff
@@ -1932,6 +1952,11 @@ declare namespace Stripe {
             description: string;
             discount: coupons.IDiscount;
 
+            /**
+             * The date on which payment for this invoice is due. This value will be null for invoices where billing=charge_automatically.
+             */
+            due_date: number;
```
Comment (LinusU) [inline types/stripe/index.d.ts]: `number | null`?

> This value will be null for invoices where billing=charge_automatically.

**Comment by LinusU** [comment]
Just one small nit

File: types/stripe/index.d.ts (hunk 1 above)
Comment (LinusU) [inline types/stripe/index.d.ts]: `number | null`?

> This value will be null for invoices where billing=charge_automatically.

**Review by LinusU — CHANGES_REQUESTED** [review]
Just one small thing...

File: types/stripe/index.d.ts (hunk 1 above)
Comment (dozoisch) [reply types/stripe/index.d.ts]: Oh good catch!

**Comment by dozoisch** [comment]
@LinusU alright corrected it! and added descriptions that were missing on some field.

Also `ending_balance` was missing an `| null`

**Review by LinusU — APPROVED** [review]
Nice 👍

**Pull request merged**